rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/get/70_source_filtering.yml

```diff
@@ -62,3 +62,123 @@
   - match:   { _id:      "1"       }
   - match:   { fields.count:  [1] }
   - match:   { _source.include.field1: v1 }
+
+---
+"Vector fields are included in source by default":
+  - requires:
+      cluster_features: "search.hide_vectors_in_source"
+      reason: vector _source filtering added in 8.16.0
+
+  - do:
+      indices.create:
+        index: vector-test
+        body:
+          mappings:
+            properties:
+              sparse:
+                type: sparse_vector
+              dense:
+                type: dense_vector
+                dims: 3
+
+  - do:
+      index:
+        index: vector-test
+        id: 1
+        body:
+          sparse:
+            running: 2.4097164
+            good: 2.170997
+            run: 2.052153
+          dense: [ 1.0, 2.0, 3.0 ]
+        refresh: true
+
+  - do:
+      get:
+        index: vector-test
+        id: 1
+
+  - exists: _source.sparse
+  - exists: _source.dense
+
+---
+"Explicitly include vector fields in source":
+  - requires:
+      cluster_features: "search.hide_vectors_in_source"
+      reason: vector _source filtering added in 8.16.0
+
+  - do:
```

Comment:
Nitpick: You could create this index as a setup step and re-use it instead of creating it for each test case.

Reply:
Could we condition the setup on a cluster feature without conditioning the entire test suite on it?